### PR TITLE
turbovnc: 2.2.7 -> 3.0, unvendor libs

### PR DESCRIPTION
**Pull request by wamserma**
###### Things done

- Built on platform(s)
  - [X] x86_64-linux
  - [x] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [X] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.05 Release Notes (or backporting 21.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2205-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by wamserma** [comment]
@ofborg eval

**Review by tpwrules — CHANGES_REQUESTED** [review]
I tried this running as a server on `x86_64-linux` and TurboVNC 3.0 (on macOS, not from nixpkgs) as a client, along with VirtualGL 3.0, and everything worked great.

Your PR title and commit message have a Unicode character in the middle that doesn't render on my system instead of the expected ASCII `->`. Please fix these so the history doesn't get polluted.

I think it would also be nice to add `xterm` to the server path on line 117 to give the user something other than a black screen when they connect. The total addition to the closure size is 2.63MiB out of 1.43GiB. While you are at it, combine line 124 to avoid double-wrapping.

File: pkgs/tools/admin/turbovnc/default.nix [hunk 1]
```diff
@@ -14,21 +17,24 @@
 , openssl
 , pam
 , perl
+, python3
 , which
 , xkbcomp
 , xkeyboard_config
 , xorg
+, xterm
```
Comment (tpwrules) [inline pkgs/tools/admin/turbovnc/default.nix:25]: This doesn't actually need to be a `buildInput`.

**Review by tpwrules — APPROVED** [review]

**Review by nh2 — APPROVED** [review]
I have tested this, it works well for my use cases on Linux.

* 3.0 client against 3.0 server works
* 2.2.6 client against 3.0 server works
* 3.0 client against 2.2.6 server works

The upgrade also fixes a key annoying issue with the client in pre-3.0: That there area always scrollbars right and bottom, cutting off the outermost pixels, no matter what size you use (see attached screenshots).

Before:

image

After:

image

**Pull request merged**